onnxruntime/core/providers/cpu/math/matmul.h

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include "core/framework/op_kernel.h"
+
+namespace onnxruntime {
+
+template <typename T>
+class MatMul : public OpKernel {
+ public:
+  MatMul(const OpKernelInfo& info) : OpKernel(info) {}
+
+  Status Compute(OpKernelContext* context) const override;
+};
+
+#if !defined(USE_MKLML_FOR_BLAS)
```

Comment:
To limit the duplication of code here, you could wrap just the PrePack with the !USE_MKLML_FOR_BLAS. If the buffer is never packed, then it doesn't matter that packed_b_ is always empty for the MKLML case, it will then go to MKL. You could look at the quantized GEMM to see how I conditionally disabled all of the prepack code for non-x64 builds.